src/event/event.rs

```diff
@@ -190,16 +190,24 @@ impl Event {
 
 impl fmt::Debug for Event {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("Event")
-            .field("token", &self.token())
+        let alternate = f.alternate();
+        let d = &mut f.debug_struct("Event");
+        d.field("token", &self.token())
             .field("readable", &self.is_readable())
             .field("writable", &self.is_writable())
             .field("error", &self.is_error())
             .field("read_closed", &self.is_read_closed())
             .field("write_closed", &self.is_write_closed())
             .field("priority", &self.is_priority())
             .field("aio", &self.is_aio())
-            .field("lio", &self.is_lio())
-            .finish()
+            .field("lio", &self.is_lio());
+
+        if alternate {
+            let mut data_str = String::new();
```

Comment:
Can we find a better way to do this then allocating a string?

Reply:
Would be better if we can somehow pass `fmt::Formatter`.

Reply:
Redone by passing the fmt::Formatter.